test/e2e/eraser_test.go

```diff
@@ -187,8 +126,168 @@ func TestRemoveImagesFromAllNodes(t *testing.T) {
+			}
+			ctx = context.WithValue(ctx, redis, &redisDep)
+
+			caddyDep := appsv1.Deployment{
+				ObjectMeta: metav1.ObjectMeta{Name: caddy, Namespace: cfg.Namespace()},
+			}
+			if err = wait.For(conditions.New(client.Resources()).DeploymentConditionMatch(&caddyDep, appsv1.DeploymentAvailable, corev1.ConditionTrue),
+				wait.WithTimeout(time.Minute*1)); err != nil {
+				t.Fatal("caddy deployment not found", err)
+			}
+			ctx = context.WithValue(ctx, caddy, &caddyDep)
+
+			return ctx
+		}).
+		Assess("Remove some of the deployments so the images aren't running", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			// Here we remove the rdis and caddy deployments
+			// Keep nginx running and ensure nginx is not deleted.
+			var redisPods corev1.PodList
+			if err := cfg.Client().Resources().List(ctx, &redisPods, func(o *metav1.ListOptions) {
+				o.LabelSelector = labels.SelectorFromSet(map[string]string{"app": redis}).String()
+			}); err != nil {
+				t.Fatal(err)
+			}
+			if len(redisPods.Items) != 2 {
+				t.Fatal("missing pods in redis deployment")
+			}
+
+			var caddyPods corev1.PodList
+			if err := cfg.Client().Resources().List(ctx, &caddyPods, func(o *metav1.ListOptions) {
+				o.LabelSelector = labels.SelectorFromSet(map[string]string{"app": caddy}).String()
+			}); err != nil {
+				t.Fatal(err)
+			}
+			if len(caddyPods.Items) != 2 {
+				t.Fatal("missing pods in caddy deployment")
+			}
+
+			err := cfg.Client().Resources().Delete(ctx, ctx.Value(redis).(*appsv1.Deployment))
+			if err != nil {
+				t.Fatal(err)
+			}
+			err = cfg.Client().Resources().Delete(ctx, ctx.Value(caddy).(*appsv1.Deployment))
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			err = wait.For(conditions.New(cfg.Client().Resources()).ResourcesDeleted(&redisPods), wait.WithTimeout(time.Minute*1))
+			if err != nil {
+				t.Fatal(err)
+			}
+			err = wait.For(conditions.New(cfg.Client().Resources()).ResourcesDeleted(&caddyPods), wait.WithTimeout(time.Minute*1))
+			if err != nil {
+				t.Fatal(err)
+			}
+			return ctx
+		}).
+		Assess("All non-running images are removed from the cluster", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			imgList := &eraserv1alpha1.ImageList{
+				ObjectMeta: metav1.ObjectMeta{Name: prune},
+				Spec: eraserv1alpha1.ImageListSpec{
+					Images: []string{"*"},
+				},
+			}
+
+			if err := cfg.Client().Resources().Create(ctx, imgList); err != nil {
+				t.Fatal(err)
+			}
+			ctx = context.WithValue(ctx, prune, imgList)
+
+			// The first check could take some extra time, where as things should be done already for the 2nd check.
+			// So we'll give plenty of time and fail slow here.
+			ctxT, cancel := context.WithTimeout(ctx, 5*time.Minute)
+			defer cancel()
+			checkImageRemoved(ctxT, t, getClusterNodes(t), redis)
+
+			ctxT, cancel = context.WithTimeout(ctx, time.Minute)
+			defer cancel()
+			checkImageRemoved(ctxT, t, getClusterNodes(t), caddy)
+
```

Comment:
do we want to check if nginx is still there?

Reply:
Done.